tests/acceptance/change.spec.js

```diff
@@ -0,0 +1,176 @@
+'use strict';
+
+let expect = require('chai').expect;
+let path = require('path');
+let change = require('../../addon/ng2/utilities/change');
+let mockFs = require('mock-fs');
+let fs = require('fs');
+let fsPromise = require('../../addon/ng2/utilities/fs-promise');
+
+
+describe('InsertChange: adding string to the source code', () => {
+  let sourcePath = 'src/app/my-component';
+  beforeEach(() => {
+    let mockDrive = {
+      'src/app/my-component': {
+        'component-file.txt': 'hello'
+      }
+    };
+    mockFs(mockDrive);
+  });
+
+  afterEach(() => {
+    mockFs.restore();
+  });
+
+  it('should add text to the source code', () => {
+    let sourceFile = path.join(sourcePath, 'component-file.txt');
+    expect(fs.existsSync(sourceFile)).to.equal(true);
+    let changeInstance = new change.InsertChange(sourceFile, 6, ' world!');
+    return changeInstance
+      .apply()
+      .then(() => {
+        return fsPromise.readWithPromise(sourceFile);
+      }).then(contents => {
+        let expectedContents = 'hello world!';
+        expect(contents).to.equal(expectedContents);
+      })
+  })
+
+  it('should add text to the source code but with unexpected output', () => {
+    let sourceFile = path.join(sourcePath, 'component-file.txt');
+    expect(fs.existsSync(sourceFile)).to.equal(true);
+    let changeInstance = new change.InsertChange(sourceFile, -6, ' world!');
+    return changeInstance
+      .apply()
+      .then(() => {
+        return fsPromise.readWithPromise(sourceFile);
+      }).then(contents => {
+        let expectedContents = 'hello world!';
+        expect(contents).to.not.equal(expectedContents);
+      });
+  })
+
+  it('should not have any changes to the source code', () => {
+    let sourceFile = path.join(sourcePath, 'component-file.txt');
+    expect(fs.existsSync(sourceFile)).to.equal(true);
+    let changeInstance = new change.InsertChange(sourceFile, 6, '');
+    return changeInstance
+      .apply()
+      .then(() => {
+        return fsPromise.readWithPromise(sourceFile);
+      }).then(contents => {
+        let expectedContents = 'hello';
+        expect(contents).to.equal(expectedContents);
+      });
+  })
+
+
+});
+
+describe('RemoveChange: remove string from the source code', () => {
+  let sourcePath = 'src/app/my-component';
+  beforeEach(() => {
```

Comment:
Can you add an outer `describe` for the whole change system, and refactor the `beforeEach` into a `beforeEach` at that level.

Since the different *Change tests use different contents, you may need to mock multiple files.